ssl/ssl.h

```diff
@@ -447,6 +447,15 @@ EXP_FUNC int STDCALL ssl_match_fingerprint(const SSL *ssl, const uint8_t* fp);
 EXP_FUNC int STDCALL ssl_match_spki_sha256(const SSL *ssl, const uint8_t* hash);
 
 /**
+ * @brief Check if SHA1 hash of CA public key matches the one given.
+ *
+ * @param ssl [in] An SSL object reference.
+ * @param fp [in] SHA1 hash to match against
+ * @return SSL_OK if the certificate is verified.
+ */
+EXP_FUNC int STDCALL ssl_match_auth_key_sha1(const SSL *ssl, const uint8_t* hash);
```

Comment:
> ... authorityKeyIdentifier doesn't tell anything about authenticity of the certificate.

That was the part that I was missing. I was, wrongly, thinking that the `authorityKeyIdentifier` is integral part of the certificate, set in stone during the signing and it will always contain the SHA1 from the CA public key and cannot be faked. 

My idea with that function was to add a comparison based on the CA public key. Which should be valid for a long period of time (10 years for example). This way I don't need to rely on the SHA1 fingerprint of the certificate (which changes often (3 months ... 1 year), nor do I need to rely on the SHA256 of the public key of the certificate (which can change 1...2 years). 

Ok, the `asn1_auth_key_id` extraction is working and can be used as a base for the improved verification process that  you've suggested.